### PR TITLE
PMMI-210: added last code changes

### PR DESCRIPTION
**Pull request by h2so3-id**
Added last code changes from pmmi project.
There were changes to the existing project. Tried to transfer them to the repository.
Please take a look. @dsdobrzynski  @dd-zen

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (ghost) [inline .gitignore]: Please add a newline at the end of the file.

**Review by ghost — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (h2so3-id) [reply .gitignore]: added

**Review by ghost — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (dsdobrzynski) [inline .gitignore:1]: It's actually possible now to commit `.idea` to help with making standards adherence easier. You just have to exclude certain files. Here is what we have excluded on some of our other projects:
```
###> PhpStorm ###
.idea/**/workspace.xml
.idea/**/tasks.xml
.idea/dictionaries
.idea/httpRequests
# Sensitive or high-churn files:
.idea/**/dataSources/
.idea/**/dataSources.ids
.idea/**/dataSources.xml
.idea/**/dataSources.local.xml
.idea/**/sqlDataSources.xml
.idea/**/dynamic.xml
.idea/**/uiDesigner.xml
# Gradle:
.idea/**/gradle.xml
.idea/**/libraries
# File-based project format:
*.iws
###< PhpStorm ###
```

**Review by dsdobrzynski — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (dsdobrzynski) [reply .gitignore:1]: @dd-zen @h2so3-id ^^
No need to fix. Just an FYI.

**Pull request merged**